hoot-core/src/main/cpp/hoot/core/io/OsmGbdxXmlWriter.cpp

```diff
@@ -0,0 +1,704 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2015, 2016, 2017, 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#include "OsmGbdxXmlWriter.h"
+
+// Boost
+using namespace boost;
+
+// Hoot
+#include <hoot/core/OsmMap.h>
+#include <hoot/core/elements/Node.h>
+#include <hoot/core/elements/Relation.h>
+#include <hoot/core/elements/Tags.h>
+#include <hoot/core/elements/Way.h>
+#include <hoot/core/filters/NoInformationCriterion.h>
+#include <hoot/core/index/OsmMapIndex.h>
+#include <hoot/core/schema/OsmSchema.h>
+#include <hoot/core/util/ConfigOptions.h>
+#include <hoot/core/util/Exception.h>
+#include <hoot/core/util/Factory.h>
+#include <hoot/core/util/MetadataTags.h>
+#include <hoot/core/util/OsmUtils.h>
+#include <hoot/core/util/UuidHelper.h>
+#include <hoot/core/visitors/CalculateMapBoundsVisitor.h>
+
+// Qt
+#include <QBuffer>
+#include <QDateTime>
+#include <QFile>
+#include <QXmlStreamWriter>
+
+using namespace geos::geom;
+using namespace std;
+
+namespace hoot
+{
+
+unsigned int OsmGbdxXmlWriter::logWarnCount = 0;
+
+HOOT_FACTORY_REGISTER(OsmMapWriter, OsmGbdxXmlWriter)
+
+OsmGbdxXmlWriter::OsmGbdxXmlWriter() :
+_formatXml(ConfigOptions().getOsmMapWriterFormatXml()),
+_precision(ConfigOptions().getWriterPrecision()),
+_encodingErrorCount(0),
+_fileNumber(0)
+{
+}
+
+OsmGbdxXmlWriter::~OsmGbdxXmlWriter()
+{
+  close();
+}
+
+QString OsmGbdxXmlWriter::removeInvalidCharacters(const QString& s)
+{
+  // See #3553 for an explanation.
```

Comment:
At least say what repo this issue is in!

Reply:
fixed. I think it was in redmine...